src/Servers/Kestrel/Core/src/Internal/Http3/Http3Connection.cs

```diff
@@ -198,13 +197,45 @@ public void Tick()
                 return;
             }
 
-            // It's safe to use UtcNowUnsynchronized since Tick is called by the Heartbeat.
-            var now = _systemClock.UtcNowUnsynchronized;
-            _timeoutControl.Tick(now);
+            UpdateStartingStreams();
+        }
+
+        private void UpdateStartingStreams()
+        {
+            var now = _systemClock.UtcNow.Ticks;
 
-            // TODO cancel process stream loop to update logic.
+            lock (_streams)
+            {
+                foreach (var stream in _streams.Values)
+                {
+                    if (stream.HasHeader)
```

Comment:
From Http3Stream:

```C#
public bool HasHeader => _requestHeaderParsingState >= RequestHeaderParsingState.Headers;
```

This looks to me to only guarantee header parsing has started. Does this actually mean header parsing is complete?

Reply:
Hmm, yes it could mean headers are still being parsed. Because HTTP/3 doesn't have continuation frames then we have all the data from the client. The request stream will either succeed or fail.

But, I get it could create confusion. I'll find a better thing to test on.